src/prototype_creep_resources.js

```diff
@@ -645,42 +645,34 @@ Creep.prototype.transferEnergyMy = function() {
 
 Creep.prototype.reserverSetLevel = function() {
   this.memory.level = 2;
-  if (this.room.controller.reservation && this.room.controller.reservation.ticksToEnd > 4500) {
+  if (this.room.controller.reservation && (Game.rooms[this.memory.base].energyCapacityAvailable < 1300 || this.room.controller.reservation.ticksToEnd > 4500)) {
```

Comment:
Both number are better. Either calculated via the `BODYPART_COST` constants, or from `config`